### PR TITLE
MRG: replace news20 by more flexible libsvm

### PR DESCRIPTION
**Pull request by mathurinm**

### CODE REVIEW

File: datasets/libsvm.py [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
 
 
 class Dataset(BaseDataset):
-    name = "news20"
-    is_sparse = True
+
+    name = "libsvm"
+
+    parameters = {
+        'dataset': ["news20.binary"],
```
Comment (tomMoral) [inline datasets/libsvm.py:15]: I would go for a name `news20` and have a mapping from this to `news20.binary` to keep the dataset names readable.

File: datasets/libsvm.py [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
+    }
 
     install_cmd = 'conda'
     requirements = ['pip:libsvmdata']
 
-    def __init__(self):
+    def __init__(self, dataset="news20.binary"):
+        self.dataset = dataset
         self.X, self.y = None, None
```
Comment (tomMoral) [inline datasets/libsvm.py:23]: Maybe try to set `self.name = news20` here and see if benchopt works?

**Review by tomMoral — COMMENTED** [review]
Some comments on the dataset name.

**Comment by agramfort** [comment]
I am thinking now that name is not critical for plots as it’s just in the drop down menu in results page. It’s not like solvers. Also having a full name is more explicit about the source of the data and if there variants

**Comment by mathurinm** [comment]
I agree; in libsvm only, there are for example :
```
 'rcv1.binary',
 'rcv1.binary_test',
 'rcv1.multiclass',
 'rcv1.multiclass_test',
 'rcv1_topics_test',
```
and in our case:
```
 'news20.binary',
 'news20.multiclass',
```

I'd say in that case explicit is better than implicit

**Comment by agramfort** [comment]
> I'd say in that case explicit is better than implicit

+1

>

**Review by tomMoral — APPROVED** [review]
LGTM!

Ok for explicit better than implicit.

**Pull request merged**